From e89bf7275eedeaebf90989bb80abdfe84b89a539 Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Tue, 4 Aug 2020 01:25:42 +0530
Subject: [PATCH] gio: Disable generic DNS record lookup on UWP

`DnsQuery()` is not available when building for UWP, so return an
error if DNS record lookup is requested. Lookup by name and by address
still works.
---
 gio/gthreadedresolver.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/gio/gthreadedresolver.c b/gio/gthreadedresolver.c
index 7691b91..a42d381 100644
--- a/gio/gthreadedresolver.c
+++ b/gio/gthreadedresolver.c
@@ -755,7 +755,7 @@ g_resolver_records_from_res_query (const gchar      *rrname,
     return records;
 }
 
-#elif defined(G_OS_WIN32)
+#elif defined(G_OS_WIN32) && !defined(G_WINAPI_ONLY_APP)
 
 static GVariant *
 parse_dns_srv (DNS_RECORD *rec)
@@ -939,7 +939,7 @@ do_lookup_records (GTask         *task,
                    GCancellable  *cancellable)
 {
   LookupRecordsData *lrd = task_data;
-  GList *records;
+  GList *records = NULL;
   GError *error = NULL;
 
 #if defined(G_OS_UNIX)
@@ -1007,7 +1007,7 @@ do_lookup_records (GTask         *task,
 
 #endif  /* HAVE_RES_NQUERY */
 
-#else
+#elif !defined(G_WINAPI_ONLY_APP)
 
   DNS_STATUS status;
   DNS_RECORD *results = NULL;
@@ -1018,7 +1018,9 @@ do_lookup_records (GTask         *task,
   records = g_resolver_records_from_DnsQuery (lrd->rrname, dnstype, status, results, &error);
   if (results != NULL)
     DnsRecordListFree (results, DnsFreeRecordList);
-
+#else
+  g_set_error_literal (&error, G_RESOLVER_ERROR, G_RESOLVER_ERROR_INTERNAL,
+                       "DNS record lookup is not implemented for UWP");
 #endif
 
   if (records)
-- 
2.27.0.windows.1

